graph/src/components/store.rs

```diff
@@ -1410,11 +1415,14 @@ impl EntityCache {
 
     /// Return the changes that have been made via `set` and `remove` as
     /// `EntityModification`, making sure to only produce one when a change
-    /// to the current state is actually needed
+    /// to the current state is actually needed.
+    ///
+    /// Also returns the updated `LfuCache`.
     pub fn as_modifications(
         mut self,
         store: &(impl Store + ?Sized),
-    ) -> Result<Vec<EntityModification>, QueryExecutionError> {
+    ) -> Result<(Vec<EntityModification>, LfuCache<EntityKey, Option<Entity>>), QueryExecutionError>
```

Comment:
I feel it would be easier to understand code that uses this method if we returned a struct instead of a tuple here. Expressions like
```rust
.as_modifications(...).0
```
are sufficiently obvious inside newtype implementations but don't make for a good external API. 
```rust
.as_modifications(...).modifications
.as_modifications(...).entity_cache
```
would be more readable. This is definitely on the nit-picky side though. What do you think?

Reply:
This really isn't very readable unless you destructure as `let (modifications, _) = ...`. I replaced this with a struct so the modifications and cache can be accessed by name.